pkgs/tools/system/zram-generator/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, ronn
+, systemd
+, kmod
+}:
+
+stdenv.mkDerivation rec {
+  pname = "zram-generator";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "systemd";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-n+ZOWU+sPq9DcHgzQWTxxfMmiz239qdetXypqdy33cM=";
+  };
+
+  # RFE: Include Cargo.lock in sources
+  # https://github.com/systemd/zram-generator/issues/65
+  patches = [ ./cargo.patch ];
```

Comment:
It's probably better to define a prePatch/postPatch (whichever is recommended...) that copies Cargo.lock as a text file from here, rather than defining it as a patch. It's easier to read IMO :)